department-of-veterans-affairs/next-build
# VAGOV-CMS- 9474: Remove old recurring events from View Other Times for this event dropdown

# Description
We are removing any recurring events that are in the past so the add to calendar links are relevant with current or future events only. Unit tests were updated as well to accompany this change. We also fixed a playwright test that was failing.

## Ticket
[#9474](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/9474)

## Developer Task

```[tasklist]
```

## Testing Steps
Tested this against recurring events locally. Check and see if any old dates are being shown for those type of events. 
URL for testing: /outreach-and-events/events/64661/

## QA steps
What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?

## Screenshots
Before:
![757_Military_Community_Education___DoD_Skillbridge_Connect__Peninsula__-_Newport_News__VA___Outreach_and_events___Veterans_Affairs](https://github.com/user-attachments/assets/5e710eba-5c7e-4652-95f6-5e9996804e5d)

After:
![Outreach_and_events___757_Military_Community_Education___DoD_Skillbridge_Connect__Peninsula__-_Newport_News__VA___Veterans_Affairs](https://github.com/user-attachments/assets/c120e5b2-04b7-4242-a57a-f15d75b7b521)


## Is this PR blocked by another PR?
- Add the `DO NOT MERGE` label
- Add links to additional PRs here:


# Reviewer

### Reviewing a PR

This section lists items that need to be checked or updated when making changes to this repository. 

## Standard Checks

```[tasklist]
```

## Merging an Approved Layout

When merging a layout, you must ensure that the content type has been turned on for `next-build` inside the `CMS`. This CMS flag must be turned on for editors to preview their work using the next build preview server.

Resource types (layouts) that have not been approved by design should NOT be pushed to production. Ensure that [slug.tsx](../src/pages/[[...slug]].tsx) does not include your resource type if it is not approved.

The status of layouts should be kept up to date inside [templates.md](./templates.md). This includes QA progress, development progress, etc. A link should be provided for where testing can occur.

## Merging a Non-Approved Layout

Your new resource type should not be included inside [slug.tsx](../src/pages/[[...slug]].tsx). Items added here will go into production once merged into the `main` branch. It is imperative that we do not push anything live that has not been approved.

Ensure that this layout has been added to the [templates.md](./templates.md) file with the current status of the work.
